core/src/main/scala/org/bitcoins/core/crypto/bip32/BIP32Path.scala

```diff
@@ -0,0 +1,70 @@
+package org.bitcoins.core.crypto.bip32
+
+import org.bitcoins.core.number.UInt32
+
+abstract class BIP32Path {
```

Comment:
is there a reason this isn't sealed?

Reply:
Also is there any invariants we can add in here?

Reply:
BIP44 paths are an extension to BIP32 paths (certain path fields have specific meanings they lack in BIP32). So that makes sealing difficult. 

Don't think there are any invariants. The path can be empty, and I don't believe it has a upper bound for long it can continue